### PR TITLE
Added missing instances: Semigroup Monoid Alternative

### PR DESCRIPTION
**Pull request by TimSheard**
Added a few missing instances that are there for Maybe and should be there for StrictMaybe as well.

### CODE REVIEW

File: strict-containers/src/Data/Maybe/Strict.hs [hunk 1]
```diff
@@ -125,3 +126,18 @@ strictMaybe _ f (SJust y) = f y
...
+  x <> SNothing = x
+  (SJust x) <> (SJust y) = SJust(x <> y)
+
+instance Semigroup a => Monoid (StrictMaybe a) where
+  mempty = SNothing
+
+instance Alternative StrictMaybe where
+    empty = SNothing
+    SNothing <|> r = r
+    l        <|> _ = l
```
Comment (lehins) [inline strict-containers/src/Data/Maybe/Strict.hs:143]: Indentation is off and trailing newline
```suggestion
instance Alternative StrictMaybe where
  empty = SNothing
  SNothing <|> r = r
  l        <|> _ = l

```

File: strict-containers/src/Data/Maybe/Strict.hs [hunk 2]
```diff
@@ -125,3 +126,18 @@ strictMaybe _ f (SJust y) = f y
 
 instance Default (StrictMaybe t) where
   def = SNothing
+
+
+
+instance Semigroup a => Semigroup (StrictMaybe a) where
+  SNothing <> x = x
+  x <> SNothing = x
+  (SJust x) <> (SJust y) = SJust(x <> y)
```
Comment (lehins) [inline strict-containers/src/Data/Maybe/Strict.hs:135]: Redundant parenthesis and a missing space
```suggestion
  SJust x <> SJust y = SJust (x <> y)
```

**Review by lehins — CHANGES_REQUESTED** [review]
Implementation looks good, just some minor issues with formatting

**Comment by TimSheard** [comment]
OK, ready for another review.

**Review by lehins — APPROVED** [review]
Looks great. Thank you

**Pull request merged**